### PR TITLE
(maint) packaging-as-a-gem improvements

### PR DESCRIPTION
**Pull request by mwaggett**
cherry-picked the merge commit from master

### CODE REVIEW

File: src/puppetlabs/ezbake/core.clj [hunk 1]
```diff
@@ -847,22 +847,25 @@ Additional uberjar dependencies:
 (defmethod action "build"
   [_ lein-project build-target]
   (action "stage" lein-project build-target)
+  (exec/exec "bundle" "install" :dir staging-dir)
```
Comment (demophoon) [inline src/puppetlabs/ezbake/core.clj]: Do we also need to specify a `--path` flag here as to install to a self contained gem path?

Comment (mwaggett) [reply src/puppetlabs/ezbake/core.clj]: Is that not what `:dir staging-dir` is for?

Comment (mwaggett) [reply src/puppetlabs/ezbake/core.clj]: it might not be

Comment (mwaggett) [reply src/puppetlabs/ezbake/core.clj]: I am bad at Clojure

Comment (underscorgan) [reply src/puppetlabs/ezbake/core.clj]: @mwaggett `:dir staging-dir` doesn't have anything to do with the `--path` like you have in other bundle installs, the `:dir` specifies where the command runs and `--path` is the path bundler will install things into. Probably missed that in the other PR.

File: src/puppetlabs/ezbake/core.clj [hunk 2]
```diff
@@ -847,22 +847,25 @@ Additional uberjar dependencies:
 (defmethod action "build"
   [_ lein-project build-target]
   (action "stage" lein-project build-target)
+  (exec/exec "bundle" "install" "--path" ".bundle/gems" "--binstubs" ".bundle/bin" :dir staging-dir)
```
Comment (mwaggett) [inline src/puppetlabs/ezbake/core.clj]: @underscorgan @demophoon updated (will cherry-pick up to master after this is merged).

Comment (underscorgan) [reply src/puppetlabs/ezbake/core.clj]: have you run a test build since this change?

**Pull request merged**